src/ppom-controller.ts

```diff
@@ -345,129 +361,216 @@ export class PPOMController extends BaseControllerV2<
+  async #getNewFilesForCurrentChain(): Promise<void> {
+    const { chainId, versionInfo } = this.state;
+    for (const fileVersionInfo of versionInfo) {
+      //  download all files for the current chain.
+      if (fileVersionInfo.chainId !== chainId) {
+        continue;
+      }
+
+      await this.#getFile(fileVersionInfo);
     }
+    this.#setChainIdDataFetched(chainId);
   }
 
-  /*
-   * Fetch the blob from the PPOM cdn.
+  /**
+   * Returns an array of new files that should be downloaded and saved to storage.
+   *
+   * @returns A promise that resolves to an array of new files to download and save to storage.
    */
-  async #fetchBlob(fileUrl: string): Promise<ArrayBuffer> {
-    const response = await safelyExecute(
-      async () => fetch(fileUrl, { cache: 'no-cache' }),
-      true,
+  async #getNewFilesForAllChains(): Promise<void> {
+    const {
+      chainIdCache,
+      storageMetadata,
+      versionInfo: stateVersionInfo,
+    } = this.state;
+
+    // create a map of chainId and files belonging to that chainId
+    const chainIdsFileInfoList = chainIdCache.map(
+      (chain): { chainId: string; versionInfo: PPOMFileVersion[] } => ({
+        chainId: chain.chainId,
+        versionInfo: stateVersionInfo.filter(
+          (versionInfo) =>
+            versionInfo.chainId === chain.chainId &&
+            !this.#checkFilePresentInStorage(storageMetadata, versionInfo),
+        ),
+      }),
     );
 
-    switch (response?.status) {
-      case 200: {
-        return await response.arrayBuffer();
-      }
+    // clear already scheduled fetch if any
+    if (this.#fileScheduleInterval) {
+      clearInterval(this.#fileScheduleInterval);
+    }
 
-      default: {
-        throw new Error(`Failed to fetch file with url ${fileUrl}`);
+    // build a list of files to be fetched for all networks
+    const fileToBeFetchedList: {
+      fileVersionInfo: PPOMFileVersion;
+      isLastFileOfNetwork: boolean;
+    }[] = [];
+    chainIdsFileInfoList.forEach((chainIdFileInfo) => {
+      const { chainId, versionInfo } = chainIdFileInfo;
+      versionInfo.forEach((fileVersionInfo, index) => {
+        fileToBeFetchedList.push({
+          fileVersionInfo,
+          isLastFileOfNetwork: index === versionInfo.length - 1,
+        });
+      });
+      if (versionInfo.length === 0) {
+        // set dataFetched to true for chainId
+        this.#setChainIdDataFetched(chainId);
       }
+    });
+
+    // if schedule interval is large so that not all files can be fetched in
+    // refreshInterval, reduce schedule interval
+    let scheduleInterval = this.state.fileScheduleInterval;
+    if (
+      this.state.refreshInterval / (fileToBeFetchedList.length + 1) <
+      scheduleInterval
+    ) {
+      scheduleInterval =
+        this.state.refreshInterval / (fileToBeFetchedList.length + 1);
     }
+
+    // schedule files to be fetched in intervals
+    this.#fileScheduleInterval = setInterval(() => {
```

Comment:
Is it worth breaking up this large method by moving this callback here to a new method like `onFileScheduleInterval`?

We could also potentially rename the larger method to something like `scheduleNewFileDownloadForAllChains` so it's explicit when this method resolves, the new files won't yet exist.